lib/mix/lib/mix/tasks/release.ex

```diff
@@ -0,0 +1,913 @@
+    [
+      {"elixir",
+       Path.join(bin_path, "elixir")
+       |> File.read!()
+       |> String.replace(~s[ -pa "$SCRIPT_PATH"/../lib/*/ebin], "")
+       |> replace_erts_bin(release, ~s["$SCRIPT_PATH"/../../erts-#{release.erts_version}/bin/])},
+      {"iex", File.read!(Path.join(bin_path, "iex"))}
+    ]
+  end
+
+  defp elixir_cli_for(:windows, bin_path, release) do
+    [
+      {"elixir.bat",
+       Path.join(bin_path, "elixir.bat")
+       |> File.read!()
+       |> String.replace(~s[goto expand_erl_libs], ~s[goto run])
+       |> replace_erts_bin(release, ~s["%dp0\\..\\..\\erts-#{release.erts_version}\\bin\\"])},
+      {"iex.bat", File.read!(Path.join(bin_path, "iex"))}
+    ]
+  end
+
+  defp executable!(path), do: File.chmod!(path, 0o744)
+
+  defp replace_erts_bin(contents, release, new_path) do
+    if release.erts_source do
+      String.replace(contents, ~s[ERTS_BIN=""], ~s[ERTS_BIN=#{new_path}])
+    else
+      contents
+    end
+  end
+
+  ## Templates
+
+  embed_text(:vm_args, ~S"""
+  ## Do not load code from filesystem as all modules are preloaded
+  -mode embedded
+
+  ## Disable the heartbeat system to automatically restart the VM
+  ## if it dies or becomes unresponsive. Useful only in daemon mode.
+  ##-heart
+
+  ## Number of diry schedulers doing IO work (file, sockets, etc)
+  ##+SDio 5
+
+  ## Increase number of concurrent ports/sockets
+  ##-env ERL_MAX_PORTS 4096
+
+  ## Tweak GC to run more often
+  ##-env ERL_FULLSWEEP_AFTER 10
+  """)
+
+  embed_template(:start, ~S"""
+  #!/bin/sh
+  set -e
+  # Feel free to edit this file in anyway you want
+  # To start your system using IEx: . $(dirname "$0")/<%= @name %> start iex
+  # To start it as a daemon using IEx: . $(dirname "$0")/<%= @name %> daemon iex
+  . $(dirname "$0")/<%= @name %> start
+  """)
+
+  embed_template(:cli, ~S"""
+  #!/bin/sh
+  set -e
+
+  SELF=$(readlink "$0" || true)
+  if [ -z "$SELF" ]; then SELF="$0"; fi
+  export RELEASE_ROOT="$(cd "$(dirname "$SELF")/.." && pwd -P)"
+  export RELEASE_NAME="${RELEASE_NAME:-"<%= @name %>"}"
+  export RELEASE_VSN="${RELEASE_VSN:-"$(cut -d' ' -f2 "$RELEASE_ROOT/releases/start_erl.data")"}"
+  export COOKIE=${COOKIE:-$(cat "$RELEASE_ROOT/releases/COOKIE")}
+  REL_VSN_DIR="$RELEASE_ROOT/releases/$RELEASE_VSN"
+
+  gen_id () {
+    od -t x -N 4 /dev/urandom | head -n1 | awk '{print $2}'
+  }
+
+  rpc () {
+    exec "$REL_VSN_DIR/elixir" \
+         --hidden --name "rpc-$(gen_id)@127.0.0.1" --cookie "$COOKIE" \
+         --boot "${REL_VSN_DIR}/remote" \
+         --boot-var RELEASE_LIB "$RELEASE_ROOT/lib" \
+         --rpc-eval "$RELEASE_NAME@127.0.0.1" "$1"
+  }
+
+  start () {
+    exec "$REL_VSN_DIR/$1" --no-halt \
+         --werl --name "$RELEASE_NAME@127.0.0.1" --cookie "$COOKIE" \
+         --erl-config "${REL_VSN_DIR}/sys" \
+         --boot "${REL_VSN_DIR}/start" \
+         --boot-var RELEASE_LIB "$RELEASE_ROOT/lib" \
+         --vm-args "${REL_VSN_DIR}/vm.args" "${@:2}"
```

Comment:
string indexing is undefined in POSIX sh
https://wiki.ubuntu.com/DashAsBinSh#A.24.7Bfoo:3.5B:1.5D.7D

Reply:
I see... what do you suggest? To assign $1, then call shift, and then use $@?

Reply:
The above worked. Thanks for the review on the scripts, really appreciated!

Reply:
Good idea